media/html/visjsDebuggerPanel.html

```diff
@@ -154,15 +171,32 @@
           edges.map((edge) => ({
             ...edge,
             color: defaultEdgeColor,
+            physics: isPhysicsOn(edge),
           }))
         );
         const edgesClusterStack = getEdgesClusterStack(message);
         nodes.add(message.data.addNodes);
         nodes.updateOnly(message.data.updateNodes);
-        edges.add(message.data.addEdges);
+        edges.add(
+          message.data.addEdges.map((edge) => ({
+            ...edge,
+            physics: isPhysicsOn(edge),
+          }))
+        );
         reCluster(edgesClusterStack);
       }
 
+      function isPhysicsOn(edge) {
+        let isPhysicsOn = network.body.nodes[edge.to].options.physics && network.body.nodes[edge.from].options.physics;
+        const clusters = network.findNode(edge.to).concat(network.findNode(edge.from));
+        clusters.forEach((nodeId) => {
+          if (!network.body.nodes[nodeId].options.physics) {
+            isPhysicsOn = false;
+          }
+        });
+        return isPhysicsOn;
```

Comment:
```suggestion
        if (!network.body.nodes[edge.to].options.physics || !network.body.nodes[edge.from].options.physics) {
          return false;
        }
        const clusters = network.findNode(edge.to).concat(network.findNode(edge.from));
        return clusters.every((nodeId) => network.body.nodes[nodeId].options.physics);
```

No need to go through the whole function if the first check already returns false.
No need to go through all clusters if one returns false.